litespi/__init__.py

```diff
@@ -47,20 +48,26 @@ def __init__(self, phy, with_mmap=True, with_master=True, mmap_endianness="big")
             CSRField("mux_sel", size=1, offset=0, description="SPI PHY multiplexer bit (0=SPIMMAP module attached to PHY, 1=SPI Master attached to PHY)")
         ])
 
-        self.submodules.mmap   = mmap   = LiteSPIMMAP(mmap_endianness) if with_mmap else LiteSPICore()
-        self.submodules.master = master = LiteSPIMaster() if with_master else LiteSPICore()
+        self.submodules.mmap     = mmap     = LiteSPIMMAP(mmap_endianness) if with_mmap else LiteSPICore()
+        self.submodules.master   = master   = LiteSPIMaster() if with_master else LiteSPICore()
+        self.submodules.crossbar = crossbar = LiteSPICrossbar(self._cfg.fields.mux_sel)
 
         if with_mmap:
             self.bus = mmap.bus
 
+        port_mmap, mmap_cs = crossbar.get_port(MMAP_PORT)
```

Comment:
I think that we should only get the ports and do all the other associated work for features that user requested instead of connecting dummy cores that will be optimized out. I think that will greatly improve readability.